### ISSUE TITLE
How to control keyboard navigation from an ItemDetailsRenderer + ComponentColumn?

### ISSUE THREAD

**Issue by tarekoraby** [vaadin/vaadin-grid#2056]
### Description
If one has a Grid with an ItemDetailsRenderer and a ComponentColumn, then keyboard navigation doesn't work as expected. 

To reproduce (Vaadin version  14.3.7), based on the following view:

```
@Route("")
public class GridView extends VerticalLayout {

    Grid<Person> grid;
    List<Person> personList;

    public GridView () {
        personList = new ArrayList<>();

        personList.add(new Person("Lucas"));
        personList.add(new Person("Peter"));
        personList.add(new Person("Samuel"));

        grid = new Grid<>(Person.class);
        grid.setItems(personList);

        grid.setDetailsVisibleOnClick(true);
        grid.setSelectionMode(Grid.SelectionMode.NONE);
        grid.setItemDetailsRenderer(
                new ComponentRenderer<>(person -> {
                    TextField tf = new TextField();
                    tf.setValue(person.firstName);
                    return tf;
                }));

        grid.addComponentColumn(item -> new Button("No op"))
                .setHeader("Component column");

        add(grid);
    }

    public class Person {
        private String firstName;

        public Person(String firstName) {
            setFirstName(firstName);
        }

        public String getFirstName() {
            return firstName;
        }

        public void setFirstName(String firstName) {
            this.firstName = firstName;
        }
    }
}
```

Run the view in the browser -> click on the details of the first row -> and click on the TextField therein -> Now press the Tab key.

### Expected outcome
Pressing the Tab key would continue giving the focus to the Grid (allowing further keyboard navigation with Tabs, arrows, etc.)

### Actual outcome
The focus is given to the component in the next row (i.e. the Button), and using keyboard navigation is no longer possible for the Grid.

**Comment by tomivirkki** [comment]
Hi, when you have the native focus on one of the focusable elements inside a grid cell (including the details cell), you're operating in what's called "interaction mode". If you want to go back to "navigation mode" (where you navigate with arrows etc), you can hit esc or enter. To enter interaction mode again, you can hit enter on a cell that has focusable elements.
